src/otaclient/main.py

```diff
@@ -181,3 +245,27 @@ def main() -> None:  # pragma: no cover
             )
             time.sleep(SHUTDOWN_AFTER_API_SERVER_EXIT)
             return _on_shutdown()
+
+        if client_update_control_flags.request_shutdown_event.is_set():
+            return _on_shutdown()
+
+        if client_update_control_flags.start_dynamic_client_event.is_set():
+            logger.info("request to start a new client")
+            client_update_control_flags.start_dynamic_client_event.clear()
+
+            dynamic_client_thread = Thread(
+                target=_thread_dynamic_client,
+                args=(client_update_control_flags,),
+                daemon=True,
+            )
+            dynamic_client_thread.start()
+
+            # Create a closure with the current thread instance to avoid the loop variable binding issue
+            def create_thread_exit_handler(thread):
+                def _thread_exit():
+                    thread.join(6)
+
+                return _thread_exit
+
+            # Register the atexit handler with the current thread instance
+            atexit.register(create_thread_exit_handler(dynamic_client_thread))
```

Comment:
In the existing design, each process is handled in main process.
Thus suggest to pass Event from core to main process, then handle each process based on the Event in main process.